### PR TITLE
[FLINK-35143][pipeline-connector][mysql] Expose newly added tables capture in mysql pipeline connector.

### PR DESCRIPTION
**Pull request by qg-lin**
https://issues.apache.org/jira/browse/FLINK-35143

### CODE REVIEW

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MysqlPipelineNewlyAddedTableITCase.java [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+import static org.apache.flink.cdc.connectors.mysql.source.MySqlDataSourceOptions.TABLES;
+import static org.apache.flink.cdc.connectors.mysql.source.MySqlDataSourceOptions.USERNAME;
+import static org.apache.flink.cdc.connectors.mysql.testutils.MySqSourceTestUtils.TEST_PASSWORD;
+import static org.apache.flink.cdc.connectors.mysql.testutils.MySqSourceTestUtils.TEST_USER;
+import static org.apache.flink.cdc.connectors.mysql.testutils.MySqSourceTestUtils.fetchResults;
+import static org.apache.flink.cdc.connectors.mysql.testutils.MySqSourceTestUtils.getServerId;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** IT tests to cover various newly added tables during capture process in pipeline mode. */
+public class MysqlPipelineNewlyAddedTableITCase extends MySqlSourceTestBase {
```
Comment (morazow) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MysqlPipelineNewlyAddedTableITCase.java:97]: Thanks @qg-lin for the PR!

Could you please use Junit5 in the IT test?

**Review by morazow — CHANGES_REQUESTED** [review]

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MysqlPipelineNewlyAddedTableITCase.java (hunk 1 above)
Comment (qg-lin) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MysqlPipelineNewlyAddedTableITCase.java:97]: Another PR in progress for Junit5.

**Review by morazow — APPROVED** [review]

**Comment by ruanhang1993** [comment]
@qg-lin Thanks for your PR.
Please rebase the master branch and resolve the conflicts. Then let's run the CI.

If you need helps, please be free to notice me. Thanks.

**Comment by qg-lin** [comment]
> @qg-lin Thanks for your PR. Please rebase the master branch and resolve the conflicts. Then let's run the CI.
> 
> If you need helps, please be free to notice me. Thanks.

Thanks for your review, I've resolved it.

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/MySqlDataSourceOptions.java [hunk 2]
```diff
@@ -224,6 +224,14 @@ public class MySqlDataSourceOptions {
                                     + "FLIP-147: Support Checkpoints After Tasks Finished. The flink version is required to be "
                                     + "greater than or equal to 1.14 when enabling this feature.");
 
+    @Experimental
+    public static final ConfigOption<Boolean> SCAN_NEWLY_ADDED_TABLE_ENABLED =
+            ConfigOptions.key("scan.newly-added-table.enabled")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "Whether capture the scan the newly added tables or not, by default is false. This option is only useful when we start the job from a savepoint/checkpoint.");
```
Comment (morazow) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/MySqlDataSourceOptions.java:233]: Could we drop the `capture` here?

For example:

```
Whether to scan the newly added ...
```

With both `capture` and `scan` it seems hard to understand what the flag enables.

Comment (morazow) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/MySqlDataSourceOptions.java:233]: ❓ 

```suggestion
                            "Whether to scan the newly added tables or not, by default is false. This option is only useful when we start the job from a savepoint/checkpoint.");
```

**Review by morazow — CHANGES_REQUESTED** [review]
Thanks @qg-lin 👍 

Minor wording suggestion from my side, otherwise looks good!

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/MySqlDataSourceOptions.java (hunk 2 above)
Comment (qg-lin) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/MySqlDataSourceOptions.java:233]: it's copy from source connector...

**Comment by morazow** [comment]
Got it 👍 I missed that there is doc for the flag, the doc version was already understandable.

Looks good!

**Review by morazow — APPROVED** [review]

**Review by ruanhang1993 — APPROVED** [review]
LGTM

**Comment by ruanhang1993** [comment]
@qg-lin It seems that some tests failed. Please take a look at them. Thanks.

**Comment by qg-lin** [comment]
> @qg-lin It seems that some tests failed. Please take a look at them. Thanks.

@ruanhang1993 Done, please trigger again.

**Pull request merged**